module/behaviour/skills/HeadBehaviourSoccer/src/HeadBehaviourSoccer.cpp

```diff
@@ -340,11 +343,10 @@ namespace behaviour {
                     });
         }
 
-        std::vector<Ball> HeadBehaviourSoccer::getFixationObjects(std::shared_ptr<const std::vector<Ball>> vballs,
-                                                                  bool& search) {
+        Balls HeadBehaviourSoccer::getFixationObjects(std::shared_ptr<const Balls> vballs, bool& search) {
```

Comment:
This function should probably take a `std::vector<const Balls>&` and dereference the pointer on passing in